service/catter-account-service/models/account.go

```diff
@@ -1,54 +1,151 @@
+	ID      string
+	Name    string
+	Email   string
+	Picture string
 }
 
 // CreateAccount creates an Account instance.
-func CreateAccount(acc Account) error {
-	ctx, _ := context.WithTimeout(context.Background(), 10*time.Second)
-	collection := db.GetClient().Database("catter").Collection("accounts")
-	accMarshal, _ := bson.Marshal(acc)
-	_, err := collection.InsertOne(ctx, accMarshal)
-	return err
+func CreateAccount(acc Account) (Account, error) {
+	if acc.ID == "" {
+		acc.ID = uuid.New().String()
+	}
+
+	accJSON, merr := dynamodbattribute.MarshalMap(acc)
+	if merr != nil {
+		fmt.Printf("Error while marshaling object %s\n", acc)
+		return Account{}, merr
+	}
+
+	input := dynamodb.PutItemInput{
+		Item:      accJSON,
+		TableName: aws.String(tableName),
+	}
+
+	_, perr := db.GetDynamoClient().PutItem(&input)
+	if perr != nil {
+		fmt.Printf("Failed to create item '%s' in the database\n", acc)
+		return Account{}, perr
+	}
+
+	return acc, nil
 }
 
 // GetAccount retrieves an Account instance by a given id.
 func GetAccount(id string) (Account, error) {
-	var acc Account
-	ctx, _ := context.WithTimeout(context.Background(), 10*time.Second)
-	collection := db.GetClient().Database("catter").Collection("accounts")
-	err := collection.FindOne(ctx, bson.D{{"id", id}}).Decode(&acc)
-	return acc, err
+	input := dynamodb.GetItemInput{
+		TableName: aws.String(tableName),
+		Key: map[string]*dynamodb.AttributeValue{
+			"ID": {
+				S: aws.String(id),
+			},
+		},
+	}
+
+	output, gerr := db.GetDynamoClient().GetItem(&input)
+	if gerr != nil {
+		fmt.Printf("There was an error retrieving item with id '%s'\n", id)
+		return Account{}, gerr
+	}
+
+	acc := Account{}
+
+	merr := dynamodbattribute.UnmarshalMap(output.Item, &acc)
+	if merr != nil {
+		fmt.Printf("Failed to unmarshal map for Account with id '%s'!\n", id)
+		return acc, merr
+	}
+
+	return acc, nil
 }
 
 // UpdateAccount updates an Account instance by a given id.
-func UpdateAccount(id string, acc Account) {
-
+func UpdateAccount(id string, acc Account) (Account, error) {
+	return Account{}, errors.New("Method not implemented")
 }
 
 // DeleteAccount deletes an Account instance by a given id.
 func DeleteAccount(id string) (Account, error) {
-	ctx, _ := context.WithTimeout(context.Background(), 10*time.Second)
-	collection := db.GetClient().Database("catter").Collection("accounts")
-	acc, err := GetAccount(id)
+	acc, gerr := GetAccount(id)
+	if gerr != nil {
+		fmt.Printf("Failed to delete Account with id '%s'!\n", id)
+		return Account{}, gerr
+	}
+
+	input := dynamodb.DeleteItemInput{
+		TableName: aws.String(tableName),
+		Key: map[string]*dynamodb.AttributeValue{
+			"ID": {
+				S: aws.String(id),
+			},
+		},
+	}
+
+	_, derr := db.GetDynamoClient().DeleteItem(&input)
+	if derr != nil {
+		fmt.Printf("Failed to delete Account with id '%s'\n", id)
+		return Account{}, derr
+	}
+
+	return acc, nil
+}
+
+// InitAccountsTable initializes the Accounts table.
+func InitAccountsTable() {
```

Comment:
This shouldn't happen here.

This should be part of the infrastructure setup, i.e Terraform modules.